rasa_core/policies/ensemble.py

```
@@ -10,84 +10,89 @@
-        if not training_data.is_empty():
+    def train(self, training_trackers, training_events, domain, **kwargs):
+        # type: (List[DialogueStateTracker], Dict, Domain, **Any) -> None
+        if training_trackers:
             for policy in self.policies:
-                policy.prepare(featurizer,
-                               max_history=training_data.max_history())
-                policy.train(training_data, domain, **kwargs)
-                self.training_metadata.update(training_data.metadata)
+                policy.train(training_trackers, domain, **kwargs)
+            self.training_events = training_events
         else:
             logger.info("Skipped training, because there are no "
                         "training samples.")
 
+    def probabilities_using_best_policy(self, tracker, domain):
+        # type: (DialogueStateTracker, Domain) -> List[float]
+        raise NotImplementedError
+
     def predict_next_action(self, tracker, domain):
-        # type: (DialogueStateTracker, Domain) -> (float, int)
+        # type: (DialogueStateTracker, Domain) -> int
         """Predicts the next action the bot should take after seeing x.
 
         This should be overwritten by more advanced policies to use ML to
         predict the action. Returns the index of the next action"""
         probabilities = self.probabilities_using_best_policy(tracker, domain)
-        max_index = np.argmax(probabilities)
+        max_index = int(np.argmax(probabilities))
         logger.debug("Predicted next action '{}' with prob {:.2f}.".format(
                 domain.action_for_index(max_index).name(),
                 probabilities[max_index]))
         return max_index
 
-    def probabilities_using_best_policy(self, tracker, domain):
-        raise NotImplementedError
+    def _max_histories(self):
+        # type: () -> List[Optional[int]]
+        """Return max history, only works if the ensemble is already trained."""
```

can we raise an exception if it is not?

---

if it is not = if the ensemble is trained?

---

that's actually wrong description, it would work before training as well